docs/oci_registries_crash_course.md

```diff
@@ -0,0 +1,115 @@
+be stored only once, and the address of the blob will be the same. This is achieved
+by using the [digest](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the blob as the address.
+
+This is the actual mechanism for allowing images to be optimal in terms of storage.
+Since most images begin by depending on the same base image, this means that
+it can be stored only once and referenced by multiple images.
+
+### Manifests
+
+A manifest is just a JSON document. It is stored seperately from the blobs.
+But similarly to the blobs, it is also content addressable. The manifest gets
+formated to a canonical form and then hashed. The hash becomes the address of
+the manifest.
+
+Where manifests become powerful is when they are used to reference to other
+manifests or blobs. These references are called [OCI Content Descriptors](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md).
+For example, the Image Manifest that most readers would be familiar with containes
+a lisf of layers that are used to build the image. Each layer is a reference to
+a blob.
+
+An additional reference outside the layer references is used for storing 
+additional configuration. More info can be found in the official [OCI Image Manifest spec](https://github.com/opencontainers/image-spec/blob/v1.0.1/manifest.md).
+
+The term _artifact_ is just a generalization over the Image Manifest idea.
+The Image Manifest has a specific format that can be found in the aforementioned [OCI Image Manifest spec](https://github.com/opencontainers/image-spec/blob/v1.0.1/manifest.md).
+The important field is the .config.mediaType field. Based on the official guidelines,
+it can be used for defining types other than an image. More info can be found [here](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage).
+
+Combining all of these ideas, we can see that artifacts can be represented by 1 
+specific object - a manifest. That manifest, based on the media type in the config, can be 
+interpreted by different programs for their specific use case. Eg. a Helm chart.
+Helm charts specifically have a mediaType of `application/vnd.cncf.helm.config.v1+json`.
+Layers, too, have a media type. The media type of a layer when the artifact
+is a helm chart is one of:
+- application/vnd.cncf.helm.config.v1+json
+- application/vnd.cncf.helm.chart.content.v1.tar+gzip
+- application/vnd.cncf.helm.chart.provenance.v1.prov
+
+More info specifically on Helm charts as OCI Artifacts can be found here: [Helm OCI MediaTypes](https://helm.sh/blog/helm-oci-mediatypes/).
+
+### Tags
+
+While manifests and blobs being content addressable is nice, it becomes hard
+to address them in a human-readable format. Tags allow us to add a reference
+to a given manifest that differs from the digest of the object.
+
+Normally, whenever we push an image to a registry, if we don't give it an
+explicit tag, the registry would most likey give it the tag `latest`. This,
```

Comment:
Is automatically tagging with `latest` in the spec? I am not aware of registry that is doing this, usually it is a client side configuration (mainly on the CI instruments building and pushing the images to the registries).
It should not be required each image to have a tag, this is why I find it unexpected the registry to tag untagged image.


```suggestion
Normally, whenever we push an image to a registry, if we don't give it an
explicit tag, the registry would most likely give it the tag `latest`. This,
```